src/plugins/graph/adornments/connecting-lines/connecting-lines.tsx

```diff
@@ -37,6 +38,15 @@ interface IConnectLines {
 export const ConnectingLines = observer(function ConnectingLines({dotsRef}: IConnectLines) {
   const graphModel = useGraphModelContext();
   const foundLinePoints = usePointLocations();
+  // access the axis domains so that a render will be triggered when they change
+  const xAxis = graphModel.getAxis("bottom");
+  const xDomain = isNumericAxisModel(xAxis) ? xAxis.domain : undefined;
+  const yAxis = graphModel.getAxis("left");
+  const yDomain = isNumericAxisModel(yAxis) ? yAxis.domain : undefined;
+  const domains = { xDomain, yDomain }; // eslint-disable-line unused-imports/no-unused-vars
+
+  // TODO: these are essentially side effects which should be handled in a MobX reaction
+  // installed by a useEffect() (which should also improve the render lag currently present).
```

Comment:
I'd be interested to talk through what this would look like. I don't have a good understanding of how that differs from an observer.